users/src/profile/index.ts

```diff
@@ -0,0 +1,21 @@
+import { Router } from "express";
+import { Profile } from "../entities/Profile";
+
+
+const router = Router()
+
+router.post("/", async (req, res) => {
+    const em = req.orm.em.fork();
+    
+    const uid = req.firebaseToken.uid;
+    const body = req.body;
+
+    const profile = await em.upsert(Profile,
+            { uid: uid, name: body.name, preferredLang: body.preferredLang, imageUrl: body.imageUrl })
+
+    await em.persistAndFlush(profile);
+
+    res.send({ success: true });
```

Comment:
Noticed you're using await and then sending response. If this fails, then will express drop the req-res chain without response? (So should frontend make sure their timeouts are some sane 10s or sth instead of the default 30s, and catch timeouts as server error?) 

I'm personally in favour of an explicit .then and .await sequence to help w debugging (differentiating btw orm error / vs servers not handling req properly). But this is a fairly simple flow, so its fine to leave it as it is.

Reply:
Resolved by catching error and returning response